### PR TITLE
[BF16] Add BF16 support to _aminmax and _anminmax_all operators

### PR DESCRIPTION
**Pull request by yusuo**
Summary: Add BF16 support to _aminmax_all and _aminmax operators.

Test Plan:
Added unit test:
https://www.internalfb.com/intern/testinfra/testconsole/testrun/2533274857208373/

Differential Revision: D30073837

### CODE REVIEW

File: test/test_reductions.py [hunk 1]
```diff
@@ -667,7 +667,7 @@ def test_amax(self, device, dtype):
 
     @onlyOnCPUAndCUDA
     @dtypesIfCPU(torch.float, torch.double)
-    @dtypesIfCUDA(torch.half, torch.float)
+    @dtypesIfCUDA(torch.half, torch.float, torch.bfloat16)
```
Comment (anjali411) [inline test/test_reductions.py:670]: you should also extend the OpInfo tests for `aminmax` here: https://github.com/pytorch/pytorch/blob/7f1b672b7a1c4ea2b5a71db888047feff22d04af/torch/testing/_internal/common_methods_invocations.py#L6542-L6547 by updating `dtypesIfCUDA` list to include `torch.bfloat16`:

`dtypesIfCUDA=all_types_and(torch.bool, torch.float16, torch.bfloat16),`

File: aten/src/ATen/native/cuda/ReduceMinMaxKernel.cu [hunk 2]
```diff
@@ -125,7 +125,7 @@ static void aminmax_kernel_impl(
     Tensor& max_result) {
   at::TensorIterator iter = make_reduction("aminmax_cuda", min_result,
     max_result, self, dim, keepdim, self.scalar_type());
-  AT_DISPATCH_ALL_TYPES_AND2(kHalf, kBool, self.scalar_type(), "aminmax_cuda", [&]() {
+  AT_DISPATCH_ALL_TYPES_AND3(kBFloat16, kHalf, kBool, self.scalar_type(), "_aminmax_cuda", [&]() {
```
Comment (heitorschueroff) [inline aten/src/ATen/native/cuda/ReduceMinMaxKernel.cu:128]: _aminmax_cuda -> aminmax_cuda

File: aten/src/ATen/native/cuda/ReduceMinMaxKernel.cu [hunk 3]
```diff
@@ -167,7 +167,7 @@ void aminmax_allreduce_kernel_impl(const Tensor& input, Tensor& min_result, Tens
   auto iter = make_reduction("aminmax_cuda", min_result, max_result, input,
                              std::vector<int64_t>{}, false, dtype);
   TORCH_CHECK(iter.numel() > 0, "min_max on a tensor with no elements is not defined.");
-  AT_DISPATCH_ALL_TYPES_AND2(kHalf, kBool, dtype, "aminmax_cuda", [&] {
+  AT_DISPATCH_ALL_TYPES_AND3(kBFloat16, kHalf, kBool, dtype, "_aminmax_all_cuda", [&] {
```
Comment (heitorschueroff) [inline aten/src/ATen/native/cuda/ReduceMinMaxKernel.cu:170]: _aminmax_all_cuda -> aminmax_all_cuda

File: test/test_reductions.py (hunk 1 above)
Comment (yusuo) [reply test/test_reductions.py:670]: Done

**Review by anjali411 — APPROVED** [review]

File: aten/src/ATen/native/cuda/ReduceMinMaxKernel.cu (hunk 2 above)
Comment (yusuo) [reply aten/src/ATen/native/cuda/ReduceMinMaxKernel.cu]: Done

**Pull request closed without merging**